systemd/systemd.go

```diff
@@ -709,28 +766,45 @@ func (l Log) Time() (time.Time, error) {
 
 // Message of the Log, if any; otherwise, "-".
 func (l Log) Message() string {
-	if msg, ok := l["MESSAGE"]; ok {
-		return msg
+	// for MESSAGE, if there are multiple strings, just concatenate them with a
+	// newline to keep as much data from journald as possible
+	msg, err := l.parseLogRawMessageString("MESSAGE", func(stringSlice []string) (string, error) {
+		return strings.Join(stringSlice, "\n"), nil
+	})
+	if err != nil {
+		return "-"
 	}
-
-	return "-"
+	return msg
 }
 
 // SID is the syslog identifier of the Log, if any; otherwise, "-".
 func (l Log) SID() string {
-	if sid, ok := l["SYSLOG_IDENTIFIER"]; ok {
-		return sid
+	// if there are multiple SYSLOG_IDENTIFIER values, just act like there was
+	// not one, making an arbitrary choice here is probably not helpful
+	sid, err := l.parseLogRawMessageString("SYSLOG_IDENTIFIER", func([]string) (string, error) {
+		return "", fmt.Errorf("multiple identifiers not supported")
+	})
+	if err != nil || sid == "" {
```

Comment:
Would it make sense to use some other tag (e.g. "*" instead of '-') to denote "multiple identifiers not supported" error case?

Reply:
thinking about this again nearly 6 months later, this and the PID are only used in displaying the log message like this:

```
2021-01-07T20:27:06Z docker.dockerd[112562]: some message
```
where "docker.dockerd" is the SID.

in the case of no SID found we have:

```
2021-01-07T20:27:06Z -[112562]: some message
```

so I'm not sure what would look better here to use 

```
2021-01-07T20:27:06Z *[112562]: some message
```

and for the converse case of PID, we have:

```
2021-01-07T20:27:06Z docker.dockerd[-]: some message
```

and the proposal to use something else would look like:

```
2021-01-07T20:27:06Z docker.dockerd[*]: some message
```

as such I think it's best to just leave it as "-".

Reply:
FWIW journal does not print anything in such case:
```
import syslog
syslog.openlog("")
syslog.syslog("foo")
```
produces:
```
Jan 08 09:16:45 galeon [13860]: foo
```